libs/shared/form-fields/src/lib/FileUploadController/index.tsx

```diff
@@ -100,15 +100,23 @@ export const FileUploadController: FC<FileUploadControllerProps> = ({
   const [state, dispatch] = useReducer(reducer, initialUploadFiles)
 
   useEffect(() => {
-    const onlyUploadedFiles = state.filter(
-      (f: UploadFile) => f.key && f.status === 'done',
+    const uploadingFiles = state.filter(
+      (f: UploadFile) => f.status === 'uploading',
     )
 
-    const uploadAnswer: UploadFileAnswer[] = onlyUploadedFiles.map(
-      transformToAnswer,
-    )
+    // If some files are still being uploaded, then no files are uploaded
+    // This is done so data schema works for multiples
+    if (uploadingFiles.length === 0) {
+      const onlyUploadedFiles = state.filter(
+        (f: UploadFile) => f.key && f.status === 'done',
+      )
 
-    setValue(id, uploadAnswer)
+      const uploadAnswer: UploadFileAnswer[] = onlyUploadedFiles.map(
+        transformToAnswer,
+      )
+
+      setValue(id, uploadAnswer)
```

Comment:
What happens if I finish uploading 1 file completely, then start uploading a new file. Can it also block the continue button somehow in that case?

Reply:
This ended up going to draft state, there where more issues with the file uploader that we are looking into with Júní.